### PR TITLE
Implemented integration with apache kafka

### PR DESCRIPTION
**Pull request by lucasMatheusNunes**

### CODE REVIEW

File: src/main/kotlin/br/com/zup/op/events/application/EventManager.kt [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
 import br.com.zup.op.events.interfaces.model.RepublishEventRequest
 import br.com.zup.op.events.interfaces.model.RepublishEventResponse
 
 interface EventManager {
+
     fun republish(request: RepublishEventRequest): RepublishEventResponse
-}
\ No newline at end of file
+
+    fun reasonList(): ArrayList<ReasonEntity>
```
Comment (sandokandias) [inline src/main/kotlin/br/com/zup/op/events/application/EventManager.kt]: listReasons(): List<ReasonEntity>

File: src/main/kotlin/br/com/zup/op/events/application/EventManager.kt [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+
+    fun listTopics(): ArrayList<TopicEntiy>
```
Comment (sandokandias) [inline src/main/kotlin/br/com/zup/op/events/application/EventManager.kt]: : List

File: src/main/kotlin/br/com/zup/op/events/application/KafkaEventManager.kt [hunk 3]
```diff
@@ -1,32 +1,68 @@
...
+
+            response.add(topic)
+        }
+
+        return response
+    }
+
+    override fun republish(request: RepublishEventRequest): RepublishEventResponse {
+
+        var idEvent = ""
```
Comment (sandokandias) [inline src/main/kotlin/br/com/zup/op/events/application/KafkaEventManager.kt]: remove

File: src/main/kotlin/br/com/zup/op/events/application/KafkaEventManager.kt [hunk 4]
```diff
@@ -1,32 +1,59 @@
...
+    }
 
-        val topics = topicRepository.findAll()
+    override fun listTopics(): List<TopicEntity> {
+        val topics = topicConsumer.listTopics()
+        return topics.map { TopicEntity(it.key) }
+    }
 
+    override fun republish(request: RepublishEventRequest): RepublishEventResponse {
+        val reasons = reasonRepository.findAll()
```
Comment (sandokandias) [inline src/main/kotlin/br/com/zup/op/events/application/KafkaEventManager.kt]: change to `this.listReasons()`

**Review by sandokandias — CHANGES_REQUESTED** [review]

File: src/main/kotlin/br/com/zup/op/events/interfaces/controller/RepublishController.kt [hunk 5]
```diff
@@ -1,16 +1,21 @@
...
+import org.springframework.web.bind.annotation.CrossOrigin
+import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PostMapping
 import org.springframework.web.bind.annotation.RequestBody
 import org.springframework.web.bind.annotation.RequestMapping
 import org.springframework.web.bind.annotation.ResponseBody
 import org.springframework.web.bind.annotation.ResponseStatus
 import org.springframework.web.bind.annotation.RestController
 
+@CrossOrigin(origins = ["http://localhost:3000", "http://localhost"]) // Enabling CORS only for the local application. We will be changing the address as soon as it is implemented for production.
```
Comment (thiagoaag) [inline src/main/kotlin/br/com/zup/op/events/interfaces/controller/RepublishController.kt]: Remove this

File: src/main/resources/db/migration/structure/V1__init.sql [hunk 6]
```diff
@@ -1,7 +1,13 @@
-CREATE TABLE topic (
-    name varchar(255) PRIMARY KEY
+CREATE TABLE event (
```
Comment (thiagoaag) [inline src/main/resources/db/migration/structure/V1__init.sql]: Rename file to -> V1.0__init.sql

**Pull request merged**